lib/block_view.go

```diff
@@ -7620,6 +7911,166 @@ func (bav *UtxoView) _connectSwapIdentity(
 	return totalInput, totalOutput, utxoOpsForTxn, nil
 }
 
+// _verifyAccessSignature verifies if the accessSignature is correct. Valid
+// accessSignature is the signed hash of (derivedPublicKey + expirationBlock)
+// in DER format, made with the ownerPublicKey.
+func _verifyAccessSignature(ownerPublicKey []byte, derivedPublicKey []byte,
+	expirationBlock uint64, accessSignature []byte) error {
+
+	// Compute a hash of derivedPublicKey+expirationBlock.
+	expirationBlockBytes := UintToBuf(expirationBlock)
```

Comment:
Do two last things for me here:
1. Validate that the derivedPublicKey length is btcec.PubKeyLenCompressed
2. Use _EncodeUint64() or whatever that function is both here and when you produce the access signature. UintToBuf is a variable-length encoding scheme, whereas _EncodeUint64() is fixed-width.

This will guarantee that the entire accessBytes has a consistent width, and an attacker can't play games by e.g. providing a shorter public key to get a different height approved...

Reply:
Lol, so much h*zing. Had to write `EncodeUint64()` in identity from scratch, but should be good now. Super unlikely anything would have happened after we verify `derivedPublicKey` length; though for sure, better safe than sorry.